compiler/damlc/daml-compiler/src/DA/Daml/Compiler/Dar.hs

```diff
@@ -105,80 +106,100 @@ buildDar service file mbExposedModules pkgName sdkVersion buildDataFiles dalfInp
               "The following modules are declared in exposed-modules but are not part of the DALF: " <>
               show (S.toList missingExposed)
       let dalf = encodeArchiveLazy pkg
```

Comment:
We already do all of the encoding and hash calculation as part of `encodeArchiveLazy` so it seems slightly wasteful to do it again in `encodePackageHash`. How about we make `encodeArchiveLazy` return a tuple of the hash and the bytes? Alternatively we could also factor out the call to `Proto.toLazyByteString` from `encodeArchiveLazy` and have it return a `ProtoLF.Archive`.